data-protocols/dsp/dsp-catalog/dsp-catalog-api/src/main/java/org/eclipse/edc/protocol/dsp/catalog/api/controller/CatalogController.java

```diff
@@ -74,39 +76,41 @@ public CatalogController(Monitor monitor, ObjectMapper mapper, IdentityService i
-        var claimToken = identityService.verifyJwtToken(tokenRepresentation, dspCallbackAddress)
-                .orElseThrow(failure -> new AuthenticationFailedException());
+        try {
+            var tokenRepresentation = TokenRepresentation.Builder.newInstance()
+                    .token(token)
+                    .build();
 
-        var expanded = jsonLdService.expand(jsonObject);
-        if (expanded.failed()) {
-            throw new InvalidRequestException(expanded.getFailureDetail());
-        }
-        var expandedJson = expanded.getContent();
-        if (!isOfExpectedType(expandedJson, DSPACE_CATALOG_REQUEST_TYPE)) {
-            throw new InvalidRequestException(format("Request body was not of expected type: %s", DSPACE_CATALOG_REQUEST_TYPE));
-        }
+            var claimToken = identityService.verifyJwtToken(tokenRepresentation, dspCallbackAddress)
+                    .orElseThrow(failure -> new AuthenticationFailedException());
+
+            var expandedJson = jsonLdService.expand(jsonObject).orElseThrow(InvalidRequestException::new);
+
+            if (!isOfExpectedType(expandedJson, DSPACE_CATALOG_REQUEST_TYPE)) {
+                throw new InvalidRequestException(format("Request body was not of expected type: %s", DSPACE_CATALOG_REQUEST_TYPE));
+            }
 
-        var message = transformerRegistry.transform(expandedJson, CatalogRequestMessage.class)
-                .orElseThrow(failure -> new InvalidRequestException(format("Request body was malformed: %s", failure.getFailureDetail())));
+            var message = transformerRegistry.transform(expandedJson, CatalogRequestMessage.class)
+                    .orElseThrow(failure -> new InvalidRequestException(format("Request body was malformed: %s", failure.getFailureDetail())));
 
-        var catalog = service.getCatalog(message, claimToken)
-                .orElseThrow(exceptionMapper(Catalog.class));
+            var catalog = service.getCatalog(message, claimToken)
+                    .orElseThrow(exceptionMapper(Catalog.class));
 
-        var catalogJson = transformerRegistry.transform(catalog, JsonObject.class)
-                .orElseThrow(failure -> new EdcException(format("Failed to build response: %s", failure.getFailureDetail())));
+            var catalogJson = transformerRegistry.transform(catalog, JsonObject.class)
+                    .orElseThrow(failure -> new EdcException(format("Failed to build response: %s", failure.getFailureDetail())));
 
-        var compacted = jsonLdService.compact(catalogJson);
-        if (compacted.failed()) {
-            throw new InvalidRequestException(compacted.getFailureDetail());
+            var compacted = jsonLdService.compact(catalogJson).orElseThrow(failure -> new EdcException(failure.getFailureDetail()));
+
+            var result = mapper.convertValue(compacted, Map.class);
+
+            return Response.status(Response.Status.OK).entity(result).build();
+        } catch (Throwable throwable) {
```

Comment:
Catching `Throwable` is generally not a good thing since that will also trap `Error` and its subclasses which should not be done. Better to catch `EdcException` or `RuntimeException` (preferably the former).

Reply:
But for protocol communication, shouldn't an error response in the correct format be returned for any problem that occurs? If we only catch e.g. `EdcException` here, for any other exception/error that may occur, a Jersey default error message would be returned to the requesting connector. That doesn't seem right.

Reply:
Not if it is an `Error` - those should not be caught.

Reply:
Changed from Throwable to Exception.